DEFEDIT-1038 Use assigned materials everywhere

ragnard-king opened the pull request:
Fixes the following outstanding rendering/material issues:

- respect default min/mag filtering settings from game.project
- render particlefxs using material shader/samplers
- render spine models using material shader/samplers
- render sprites using material samplers

https://github.com/defold/editor2-issues/issues/543

**Still TODO** (in separate issue):

- Set sampler uniforms to the correct texture. This currently works incidentally for everything but models (where we explicitly set the sampler uniforms) because GLSL sampler uniforms seem to default to texture unit 0 if not explicitly set. When multiple samplers are defined in material, we currently just pick the first one. We should do correct matching similar to the other uniforms.

ragnard-king committed:
DEFEDIT-1038 Use assigned materials everywhere

Fixes the following outstanding rendering/material issues:

- respect default min/mag filtering settings from game.project
- render particlefxs using material shader/samplers
- render spine models using material shader/samplers
- render sprites using material samplers

RagnarSvenssonKing reviewed (changes requested):
A few questions/issues. Also, reasonable to catch any of this through tests? E.g. loading a project and asserting that a specific material will be used for rendering. Can't of course be done all the way, but any reasonable coverage there?

ragnard-king committed:
DEFEDIT-1038 Review fix

- avoid NPE in parse-filter-param

ragnard-king committed:
DEFEDIT-1038 Review fix

- Assert that assigned materials are used. Checks scene-data so is not
  a full end-to-end test, but be able to catch some regressions.
